### ISSUE TITLE
[Req] Command to query the current temperature value

### ISSUE THREAD

**Issue by kRHYME7** [hyprwm/hyprsunset#3]
~Helps to make some tools around this. Like a UI with values displayed. Also, easier to index current temperature.~

~👍🏽 Much better is to have a CLI for this tool to increase, decrease the temp.~

🐒

**Comment by vaxerski** [comment]
what

**Comment by kRHYME7** [comment]
>Helps to make some tools around this. Like a UI with values displayed. Also, easier to index current temperature.

Command to query the current temperature value so we can store it as a state. By that, we can use it to output it examples for waybar.

**Comment by vaxerski** [comment]
but it's just whatever you configured hyprsunset to? why not keep track of it yourself?

**Comment by kRHYME7** [comment]
Yes, that would be an option. If one creates a script for it, they could write it to a file or a FIFO upon committing the temperature with ` hyprsunset -t 5000 > $file ` . In my opinion, it's not really optimal and would need some additional logic to be robust.

Sorry for my ignorance, but aren't there options to query the temperature value of the current running daemon? The approach mentioned might lose track of the temperature. It's not a big deal, but this would be more optimal.

I have another concern: to avoid multiple instances of hyprsunset, is the best way to do it by using killall hyprsunset and then setting the temperature? Something like this:
` killall hyprsunset; hyprsunset -t $temp` ?

**Comment by vaxerski** [comment]
yes. hyprsunse for now has no ipc

**Comment by htho1** [comment]
Can you not just make `hyprsunset -t` echo the current value? This is how I expected it to behave, and it makes the most sense in my opinion.

**Comment by ysxninja** [comment]
> Yes, that would be an option. If one creates a script for it, they could write it to a file or a FIFO upon committing the temperature with `hyprsunset -t 5000 > $file`. In my opinion, it's not really optimal and would need some additional logic to be robust.
> 
> Sorry for my ignorance, but aren't there options to query the temperature value of the current running daemon? The approach mentioned might lose track of the temperature. It's not a big deal, but this would be more optimal—hopefully without overcomplicating the logic of hyprsunset. If that’s the case, I’ll just consider the option above.
> 
> I have another concern: to avoid multiple instances of hyprsunset, is the best way to do it by using `killall hyprsunset` and then setting the temperature? Something like this: `killall hyprsunset; hyprsunset -t $temp`?
> 
> This will be helpful in creating a knob/slider GUI.

Heres a quick workaround for waybar,
### bluelighttoggle.sh
```sh bluelighttoggle.sh
#!/bin/sh

# Check if hyprsunset is running using pgrep
pid=$(pgrep -x hyprsunset)

if [ -z "$pid" ]; then
  # If not running, start hyprsunset with the specified temperature
  hyprsunset -t 5700 &
  echo "hyprsunset started with temperature 5700"
else
  # If running, kill hyprsunset
  kill "$pid"
  echo "hyprsunset stopped"
fi
```
### bluelightinfo.sh
```sh bluelightinfo.sh
#!/bin/sh

# Check if hyprsunset is running
pid=$(pgrep -x hyprsunset)

# Set temperature based on whether hyprsunset is running or not
if [ -z "$pid" ]; then
  # If hyprsunset is not running, default temperature is 6500K
  temperature=6500
else
  # If hyprsunset is running, temperature is 5700K
  temperature=5700
fi

# Format the output for Waybar with line breaks
formatted_output="Temperature: $temperature"

# Check if the temperature is greater than or equal to 6500K
if [ "$temperature" -ge 6500 ]; then
  # Output as JSON for Waybar, escaping newlines
  echo "{\"text\": \"\", \"tooltip\": \"$formatted_output\"}"
else
  # Otherwise, output a different icon for lower temperature
  echo "{\"text\": \"󰽥\", \"tooltip\": \"$formatted_output\"}"
fi
```

### Waybar
```json
  "custom/hyprsunset-temperature": {
    "format": " {} ",
    "exec": "$HOME/.local/bin/bluelightinfo.sh",
    "tooltip": true,
    "interval": 1,
    "return-type": "json",
    "on-click": "$HOME/.local/bin/bluelighttoggle.sh"
  },

```

**Comment by Dregu** [comment]
Guess what, the IPC has a simple way to query the current value. Here's how I use it from waybar through the socket: https://gist.github.com/Dregu/ddba409a837ce87af8f9e26ef90f67e9

I assume this query is also supposed to work through hyprctl, but I had some trouble switching to the -git channel so I haven't tried that way yet. At least querying seems to be undocumented in the wiki so there's a little oversight for ya.

**Comment by kRHYME7** [comment]
https://github.com/hyprwm/hyprsunset/commit/005f0fcfdcf2d335816d1922d5730602920d323b

Didn't notice this. This is enough, closing this now

**Comment by danielwerg** [comment]
```sh
hyprctl hyprsunset temperature 4000
hyprctl hyprsunset temperature # 4000
hyprctl hyprsunset identity
hyprctl hyprsunset temperature # 4000 <--- still reports old value
hyprctl hyprsunset temperature 6500
hyprctl hyprsunset temperature # 6500
``
